docs/core/PINs/PIN-16-Obfuscating-Data-Passed-to-Cloud.md

```diff
@@ -0,0 +1,161 @@
+---
+title: 'PIN-16: Obfuscating Data Passed to Cloud'
+sidebarDepth: 0
+---
+
+# PIN-16: Obfuscating Data Passed to Cloud
+
+Date: March 6, 2020
+
+Authors: Laura Lorenz, Nathan Atkins, Alex Goodman
+
+## Status
+Draft
+
+## Context
+
+Businesses that deal with customer, medical or otherwise sensitive data are concerned about sending that data outside of their controlled environment. This includes exception messages and stack traces, partial or entire log messages that contain sensitive information, and task-produced data such as Result values being sent to Prefect Cloud. 
+
+Despite these concerns, there are some users who wish to still use features from Prefect Cloud if they can feel more confident that the data Cloud stores to provide those features is non-sensitive or obfuscated from the cloud database if so.
+
+Though the hybrid model provides protection from most of the canonical surface area of this risk by isolating user code from Cloud persistence systems, some specific types of leaks are still possible and even enforced by the library today. Currently, the Prefect logger can be bypassed if users always log to their own logger instance, but this is not immediately obvious and does not prevent exceptions from being sent through the Prefect logger to Cloud since the exception is ultimately raised during the Prefect TaskRunner. In addition, any accidental usage of JSONResultHandler (including today’s hardcoded case for Parameter type tasks) emits the return value of a task to the cloud database.
+
+This PIN seeks to argue that more configuration options, including a set-it-and-forget-it global configuration, that protects the data boundary to Cloud is a valid “safe default” scenario that should be provided alongside the current safe default scenario that prioritizes convenience.
```

Comment:
This sentence is a little confusing - I think I get the gist but it might benefit from being split into two sentences.